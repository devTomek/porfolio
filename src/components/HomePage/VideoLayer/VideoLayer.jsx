import React from 'react';
import "./VideoLayer.scss";
import manTyping from "../../../assets/manTyping.mp4";
import manTypingFrame from "../../../assets/manTypingFrame.PNG"

const VideoLayer = () =>
  <div className="video-layer">
    <video poster={manTypingFrame} autoPlay muted loop width="100%">
      <source src={manTyping}/>
      Your browser does not support the video tag.
    </video>
    <div className="text-wrapper">
      <p>This awesome videos are possible because of HTML5</p>
      <p>Check all the technologies I used to create this website below</p>
    </div>
  </div>;

export default VideoLayer;