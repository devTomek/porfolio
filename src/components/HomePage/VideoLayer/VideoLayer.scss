@import '../../../variables';

.video-layer {
  color: $background-color;
  display: flex;
  align-items: center;
  justify-content: center;
  .text-wrapper {
    background-color: rgba(0, 0, 0, 0.1);
    width: 80%;
    height: 20vh;
    position: absolute;
    display: flex;
    flex-direction: column;
    justify-content: center;
    p {
      font-size: 3rem;
      text-align: center;
    }
  }
}