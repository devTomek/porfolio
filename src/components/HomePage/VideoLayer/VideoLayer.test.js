import React from 'react';
import VideoLayer from "./VideoLayer";
import { shallow } from 'enzyme';

describe('VideoLayer.jsx', () => {
  it('should render', () => {
    const wrapper = shallow(
      <VideoLayer/>
    );

    expect(wrapper).toMatchSnapshot();
  });
});