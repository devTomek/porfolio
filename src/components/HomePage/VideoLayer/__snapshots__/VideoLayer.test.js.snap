// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoLayer.jsx should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VideoLayer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="video"
        >
          <video
            autoPlay={true}
            loop={true}
            muted={true}
            poster="manTypingFrame.PNG"
            width="100%"
          >
            <source
              src="manTyping.mp4"
            />
            Your browser does not support the video tag.
          </video>
        </div>,
        <div
          className="content-wrapper"
        >
          <div
            className="content"
          >
            <p>
              This awesome videos are possible because of HTML5
            </p>
            <p>
              Check all the technologies I used to create this website below
            </p>
          </div>
        </div>,
      ],
      "className": "video-layer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <video
            autoPlay={true}
            loop={true}
            muted={true}
            poster="manTypingFrame.PNG"
            width="100%"
          >
            <source
              src="manTyping.mp4"
            />
            Your browser does not support the video tag.
          </video>,
          "className": "video",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "autoPlay": true,
            "children": Array [
              <source
                src="manTyping.mp4"
              />,
              "Your browser does not support the video tag.",
            ],
            "loop": true,
            "muted": true,
            "poster": "manTypingFrame.PNG",
            "width": "100%",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "src": "manTyping.mp4",
              },
              "ref": null,
              "rendered": null,
              "type": "source",
            },
            "Your browser does not support the video tag.",
          ],
          "type": "video",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="content"
          >
            <p>
              This awesome videos are possible because of HTML5
            </p>
            <p>
              Check all the technologies I used to create this website below
            </p>
          </div>,
          "className": "content-wrapper",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                This awesome videos are possible because of HTML5
              </p>,
              <p>
                Check all the technologies I used to create this website below
              </p>,
            ],
            "className": "content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "This awesome videos are possible because of HTML5",
              },
              "ref": null,
              "rendered": "This awesome videos are possible because of HTML5",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Check all the technologies I used to create this website below",
              },
              "ref": null,
              "rendered": "Check all the technologies I used to create this website below",
              "type": "p",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="video"
          >
            <video
              autoPlay={true}
              loop={true}
              muted={true}
              poster="manTypingFrame.PNG"
              width="100%"
            >
              <source
                src="manTyping.mp4"
              />
              Your browser does not support the video tag.
            </video>
          </div>,
          <div
            className="content-wrapper"
          >
            <div
              className="content"
            >
              <p>
                This awesome videos are possible because of HTML5
              </p>
              <p>
                Check all the technologies I used to create this website below
              </p>
            </div>
          </div>,
        ],
        "className": "video-layer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <video
              autoPlay={true}
              loop={true}
              muted={true}
              poster="manTypingFrame.PNG"
              width="100%"
            >
              <source
                src="manTyping.mp4"
              />
              Your browser does not support the video tag.
            </video>,
            "className": "video",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "autoPlay": true,
              "children": Array [
                <source
                  src="manTyping.mp4"
                />,
                "Your browser does not support the video tag.",
              ],
              "loop": true,
              "muted": true,
              "poster": "manTypingFrame.PNG",
              "width": "100%",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "src": "manTyping.mp4",
                },
                "ref": null,
                "rendered": null,
                "type": "source",
              },
              "Your browser does not support the video tag.",
            ],
            "type": "video",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="content"
            >
              <p>
                This awesome videos are possible because of HTML5
              </p>
              <p>
                Check all the technologies I used to create this website below
              </p>
            </div>,
            "className": "content-wrapper",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  This awesome videos are possible because of HTML5
                </p>,
                <p>
                  Check all the technologies I used to create this website below
                </p>,
              ],
              "className": "content",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "This awesome videos are possible because of HTML5",
                },
                "ref": null,
                "rendered": "This awesome videos are possible because of HTML5",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Check all the technologies I used to create this website below",
                },
                "ref": null,
                "rendered": "Check all the technologies I used to create this website below",
                "type": "p",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
